step/step.go

```diff
@@ -0,0 +1,174 @@
+package step
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+
+	"github.com/bitrise-io/go-steputils/v2/export"
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-xcode/v2/autocodesign/projectmanager"
+	"github.com/bitrise-io/go-xcode/xcodeproject/xcodeproj"
+)
+
+type Updater struct {
+	inputParser stepconf.InputParser
+	exporter    export.Exporter
+	logger      log.Logger
+}
+
+func NewUpdater(inputParser stepconf.InputParser, exporter export.Exporter, logger log.Logger) Updater {
+	return Updater{
+		inputParser: inputParser,
+		exporter:    exporter,
+		logger:      logger,
+	}
+}
+
+func (u Updater) ProcessConfig() (Config, error) {
+	var input Input
+	err := u.inputParser.Parse(&input)
+	if err != nil {
+		return Config{}, err
+	}
+
+	stepconf.Print(input)
+	u.logger.Println()
+
+	return Config{
+		ProjectPath:             input.ProjectPath,
+		Scheme:                  input.Scheme,
+		Target:                  input.Target,
+		Configuration:           input.Configuration,
+		BuildVersion:            input.BuildVersion,
+		BuildVersionOffset:      input.BuildVersionOffset,
+		BuildShortVersionString: input.BuildShortVersionString,
+	}, nil
+}
+
+func (u Updater) Run(config Config) (Result, error) {
+	helper, err := projectmanager.NewProjectHelper(config.ProjectPath, config.Scheme, config.Configuration)
+	if err != nil {
+		return Result{}, err
+	}
+
+	generated, err := generatesInfoPlist(helper, config.Target, config.Configuration)
+	if err != nil {
+		return Result{}, err
+	}
+
+	buildVersion := config.BuildVersion + config.BuildVersionOffset
```

Comment:
Good point. The previous implementation also accepted negative values but I think it does not make sense. And because this will be a 2.0 breaking change release I will modify the behaviour to only accept positive values. :D